packages/lib/src/interactions/utils.ts

```diff
@@ -99,3 +106,52 @@ export function clampRectangleToVis(
   const shift = newCenter.clone().sub(center);
   return [startPoint.clone().add(shift), endPoint.clone().add(shift)];
 }
+
+export function getDistinctPoints(
+  startPoint: Vector2,
+  endPoint: Vector2,
+  xDomain: Domain,
+  yDomain: Domain
+): [Vector2, Vector2] {
+  const [xMin, xMax] = xDomain;
+
+  if (Math.abs(endPoint.x - startPoint.x) < Math.abs(xMax - xMin) / 200) {
+    return [new Vector2(xMin, startPoint.y), new Vector2(xMax, endPoint.y)];
+  }
+
+  const [yMin, yMax] = yDomain;
+  if (Math.abs(endPoint.y - startPoint.y) < Math.abs(yMax - yMin) / 200) {
```

Comment:
The criteria for an "empty" rectangle is to have the width less than 1/200 of the axis coordinates. 

I tweaked this a bit and found that this value gave the best UX in my opinion.

Reply:
Wouldn't this become much simpler if `SelectionTool`'s callbacks provided the selection in HTML space in addition to data space, like `useCanvasEvents`? `getDistinctPoints` could then define a threshold in screen pixels.

Maybe `SelectionTool` could also provide the selection in world space to simplify other computations?